### ISSUE TITLE
Rename version property for Git Commit ID Maven plugin from git-commit-id-plugin.version to git-commit-id-maven-plugin.version

### ISSUE THREAD

**Issue by jcranfordupgrade** [spring-projects/spring-boot#34397]
TLDR
- [build.gradle](https://github.com/spring-projects/spring-boot/blob/v3.0.3/spring-boot-project/spring-boot-dependencies/build.gradle#L266-L272): uses the new Maven artifactid `git-commit-id-maven-plugin` ([release 5.0.0](https://github.com/git-commit-id/git-commit-id-maven-plugin/releases/tag/v5.0.0))
- [effective pom.xml](https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/3.0.3/spring-boot-dependencies-3.0.3.pom): refers to the old Maven artifactId `git-commit-id-plugin` ([release 4.9.10](https://github.com/git-commit-id/git-commit-id-maven-plugin/releases/tag/v4.9.10))

Result is confusing mix of old and new `artifactId` versus `version` naming for this dependency in the effective pom.xml:
```
<plugin>
<groupId>io.github.git-commit-id</groupId>
<artifactId>git-commit-id-maven-plugin</artifactId>
<version>${git-commit-id-plugin.version}</version>
</plugin>
```

This is the build.gradle entry from Spring Boot Dependencies 3.0.3.

https://github.com/spring-projects/spring-boot/blob/v3.0.3/spring-boot-project/spring-boot-dependencies/build.gradle#L266-L272
```
	library("Git Commit ID Plugin", "5.0.0") {
		group("io.github.git-commit-id") {
			plugins = [
				"git-commit-id-maven-plugin"
			]
		}
	}
```

This is how the version property appears in the effective pom.xml. The property name does not match the above artifactId.
```
<git-commit-id-plugin.version>5.0.0</git-commit-id-plugin.version>
```

The version property is not directly defined in the repository, so I cannot submit a PR to fix it. There is no Gradle parent either, so Spring Boot Dependencies is not inheriting the property from somewhere else. I think maybe the effective pom.xml is generated programmatically by a `generatePomFileForMavenPublication` task, but I was not able to track down where that data is coming from.

Thank you.

**Comment by tiger3768** [comment]
Hi, I am an absolute beginner and wish to contribute, can you please guide me on how to resolve this issue?
